Fetch GitHub repositories from a given owner

jjmerchante opened the pull request:
This PR includes the implementation to retrieve the list of repositories given an owner.

This implements the function to fetch the repositories in a redis queue. From GraphQL you can create a new job given an owner and an optional GitHub API token. You can obtain the list of repositories once the job has finished using its id.

jjmerchante committed:
[log] Add job id to the name of the transaction

If the context contains a job identifier, the transaction
will include it as a suffix to its name. Therefore,
it will be possible to determine when a transaction
was created by a job.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

evamillan reviewed (approved):
LGTM

sduenas commented on bestiary/core/jobs.py:
```diff
@@ -80,3 +85,49 @@ def get_jobs():
+
+
+@django_rq.job
+def github_owner_repos(ctx, owner, api_token=None):
```
The name of the job should include a verb to make clearer what it does.

sduenas commented on bestiary/core/context.py:
```diff
@@ -23,4 +23,7 @@
+BestiaryContext = collections.namedtuple(
+    'BestiaryContext', ['user', 'job_id']
+)
+BestiaryContext.__new__.__defaults__ = (None, None)
```
What should it be initialized to None?

jjmerchante replied:
When the context is created by a job, a job identifier will be included, but when it is not created by a job, by default it will be None.

This change is based on SortingHat (https://github.com/chaoss/grimoirelab-sortinghat/blob/muggle/sortinghat/core/context.py#L29)

sduenas replied:
Thanks! I completely forgot about it.

sduenas commented on bestiary/core/retrieval/github.py:
```diff
@@ -0,0 +1,83 @@
+
+    logger.info(f"GitHub owner repositories fetched; owner='{owner}'")
+
+    return repositories
```
I wonder if it would be better to convert this into a generator, so it can be processed at the same time data is processed. For the job probably it doesn't make sense but we might want to have this for the future.

sduenas reviewed (changes requested):
In general it looks ok. Check my comments below. Also, JSON files should have on their filename, the `.json` and end with an empty line (in general all files should end like that).

jjmerchante committed:
[GHowners] Include a function to retrieve repositories from an owner

This commit fetches from GitHub API all the repositories from a specific
owner include information related like the url, whether it has issues
enabled or not and if it is a fork.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

jjmerchante committed:
[job] Include a job to fetch GitHub repositories from an owner

This commit includes the implementation of a redis queue job
to fetch all the repositories from a specific owner and store
the result or the errors in the output.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

jjmerchante committed:
[schema] Include a graphql mutation to fetch GitHub repositories

This commit includes the implementation for grpahql to fetch GitHub
repositories from a given owner and returns information like the url,
whether it has issues enabled or not and if it is a fork.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

sduenas reviewed (changes requested):
Ok to the changes but json files don't end in an empty line still.

sduenas reviewed (approved):
LGTM
